feat: add `math/base/special/lucasf`

hrshya opened the pull request:
Resolves none .

## Description

> What is the purpose of this pull request?

This pull request:

-   adds `math/base/special/lucasf` .

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   progresses #649 . 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

hrshya committed:
feat: add math/base/special/lucasf

hrshya committed:
fix: fixes ci issues

hrshya committed:
fix: fixes ci issues

hrshya commented:
This is how it is being implemented in the other packages. So why is it creating a lint issue here.

Lint : Using the `new Array()` constructor is not allowed; use an array literal with push instead  stdlib/no-new-array

cc: @kgryte @Planeshifter

hrshya committed:
resolve lint errors

Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

hrshya commented:
@anandkaranubc could you please review this PR?

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/lucasf/src/addon.c:
```diff
@@ -0,0 +1,23 @@
+#include "stdlib/math/base/napi/unary.h"
+
+// cppcheck-suppress shadowFunction
+STDLIB_MATH_BASE_NAPI_MODULE_I_F( stdlib_base_lucas )
```
```suggestion
STDLIB_MATH_BASE_NAPI_MODULE_I_F( stdlib_base_lucasf )
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/lucasf/src/main.c:
```diff
@@ -0,0 +1,79 @@
+* float out = stdlib_base_lucasf( -1 );
+* // returns NaN
+*/
+float stdlib_base_lucas( const int32_t n ) {
```
```suggestion
float stdlib_base_lucasf( const int32_t n ) {
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/lucasf/examples/c/example.c:
```diff
@@ -0,0 +1,31 @@
+	float v;
+
+	for ( i = 0; i < 35; i++ ) {
+		v = stdlib_base_lucas( i );
```
```suggestion
		v = stdlib_base_lucasf( i );
```

hrshya committed:
update README.md

Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

kgryte commented:
@anandkaranubc Would you mind taking a look at this PR?

anandkaranubc commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into feat/lucasf

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/lucasf/lib/main.js:
```diff
@@ -0,0 +1,92 @@
+// MAIN //
+
+/**
+* Computes the nth Lucas number in single-precision floating-point format.
```
```suggestion
* Computes the nth Lucas number as a single-precision floating-point number.
```

Applies here and elsewhere.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/lucasf/lib/index.js:
```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+/**
+* Compute the nth Lucas number in single-precision floating-point format.
```
```suggestion
* Compute the nth Lucas number as a single-precision floating-point number.
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/lucasf/lib/native.js:
```diff
@@ -0,0 +1,74 @@
+// MAIN //
+
+/**
+* Computes the nth Lucas number in single-precision floating-point format.
```
Same comment.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/lucasf/lib/native.js:
```diff
@@ -0,0 +1,74 @@
+* // returns 18
+*
+* @example
+* var y = lucasf( -1.0 );
```
```suggestion
* var y = lucasf( -1 );
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/lucasf/include/stdlib/math/base/special/lucasf.h:
```diff
@@ -0,0 +1,40 @@
+#endif
+
+/**
+* Computes the nth Lucas number in single-precision floating-point format.
```
Same comment

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/lucasf/src/main.c:
```diff
@@ -0,0 +1,79 @@
+*
+* @example
+* float out = stdlib_base_lucasf( 1 );
+* // returns 1
```
```suggestion
* // returns 1.0f
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/lucasf/src/main.c:
```diff
@@ -0,0 +1,79 @@
+	if ( n < 0 || n > STDLIB_CONSTANT_FLOAT32_MAX_SAFE_NTH_LUCAS ) {
+		return 0.0f / 0.0f; // NaN
+	}
+	return lucasf_value[ n ];
```
```suggestion
	return lucas_value[ n ];
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/lucasf/src/main.c:
```diff
@@ -0,0 +1,79 @@
+#include "stdlib/math/base/special/lucasf.h"
+#include "stdlib/constants/float32/max_safe_nth_lucas.h"
+
+static const int32_t lucasf_value[ 35 ] = {
```
```suggestion
static const int32_t lucas_value[ 35 ] = {
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/lucasf/package.json:
```diff
@@ -0,0 +1,69 @@
+{
+  "name": "@stdlib/math/base/special/lucasf",
+  "version": "0.0.0",
+  "description": "Compute the nth Lucas number in single-precision floating-point format.",
```
Same comment

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/lucasf/package.json:
```diff
@@ -0,0 +1,69 @@
+    "special",
+    "function",
+    "fibonacci",
+    "lucas",
```
```suggestion
    "lucasf",
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/lucasf/examples/index.js:
```diff
@@ -0,0 +1,29 @@
+var v;
+var i;
+
+for ( i = 0; i < 35; i++ ) {
```
Can you change it to use `logEachMap` instead, as done in other JS example files?

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/lucasf/docs/types/index.d.ts:
```diff
@@ -0,0 +1,78 @@
+// TypeScript Version: 4.1
+
+/**
+* Computes the nth Lucas number in single-precision floating-point format.
```
Same comment

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/lucasf/docs/repl.txt:
```diff
@@ -0,0 +1,47 @@
+
+{{alias}}( n )
+    Computes the nth Lucas number in single-precision floating-point format.
```
Same comment

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/lucasf/docs/repl.txt:
```diff
@@ -0,0 +1,47 @@
+    > y = {{alias}}( 4 )
+    7
+    > y = {{alias}}( 34 )
+    12752043
```
```suggestion
    > y = {{alias}}( 35 )
    NaN
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/lucasf/benchmark/c/native/benchmark.c:
```diff
@@ -0,0 +1,136 @@
+*/
+static double benchmark( void ) {
+	double elapsed;
+	int32_t x[ 100 ];
+	double t;
```
```suggestion
	int32_t x[ 100 ];
	double elapsed;
	double t;
```

Variables are ordered according to the decreasing order of their length.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/lucasf/benchmark/c/native/benchmark.c:
```diff
@@ -0,0 +1,136 @@
+
+	t = tic();
+	for ( i = 0; i < ITERATIONS; i++ ) {
+		y = stdlib_base_lucas( x[ i%100 ] );
```
```suggestion
		y = stdlib_base_lucasf( x[ i%100 ] );
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/lucasf/benchmark/c/native/benchmark.c:
```diff
@@ -0,0 +1,136 @@
+	double elapsed;
+	int32_t x[ 100 ];
+	double t;
+	double y;
```
```suggestion
	float y;
```

The return type is `float` here

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/lucasf/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,151 @@
+* @return elapsed time in seconds
+*/
+static double benchmark( void ) {
+	double elapsed;
+	double t;
+	int x[ 100 ];
+	int y;
+	int i;
```
```suggestion
	double elapsed;
	int x[ 100 ];
	double t;
	int y;
	int i;
```

Same comment

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/lucasf/benchmark/benchmark.js:
```diff
@@ -0,0 +1,306 @@
+		'dtype': 'float32'
+	});
+
+	function lucasf( n ) {
```
```suggestion
		// TODO: replace with `powf` when available
```

hrshya replied:
> 

@anandkaranubc I think you accidentally removed the function implementation here and I didn't double checked it.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/lucasf/README.md:
```diff
@@ -0,0 +1,252 @@
+
+# Lucasf
+
+> Compute the nth [Lucas number][lucas-number] in single-precision floating-point format.
```
```suggestion
> Compute the nth [Lucas number][lucas-number] as a single-precision floating-point number.
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/lucasf/README.md:
````diff
@@ -0,0 +1,252 @@
+2, 1, 3, 4, 7, 11, 18, 29, 47, 76, 123, 199, 322, \ldots
+```
+
+<!-- <div class="equation" align="center" data-raw-text="2, 1, 3, 4, 7, 11, 18, 29, 47, 76, 123, 199, 322, \ldots" data-equation="eq:lucas_sequence">
+    <img src="https://cdn.jsdelivr.net/gh/stdlib-js/stdlib@bb29798906e119fcb2af99e94b60407a270c9b32/lib/node_modules/@stdlib/math/base/special/lucasf/docs/img/equation_lucas_sequence.svg" alt="Lucas sequence">
+    <br>
+</div> -->
````
You can remove this, as this will be automatically populated.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/lucasf/README.md:
````diff
@@ -0,0 +1,252 @@
+L_n = \begin{cases}2 & \textrm{if}\ n = 0\\1 & \textrm{if}\ n = 1\\L_{n-1} + L_{n-2} & \textrm{if}\ n > 1\end{cases}
+```
+
+<!-- <div class="equation" align="center" data-raw-text="L_n = \begin{cases}2 &amp; \textrm{if}\ n = 0\\1 &amp; \textrm{if}\ n = 1\\L_{n-1} + L_{n-2} &amp; \textrm{if}\ n &gt; 1\end{cases}" data-equation="eq:lucas_recurrence_relation">
+    <img src="https://cdn.jsdelivr.net/gh/stdlib-js/stdlib@bb29798906e119fcb2af99e94b60407a270c9b32/lib/node_modules/@stdlib/math/base/special/lucasf/docs/img/equation_lucas_recurrence_relation.svg" alt="Lucas sequence recurrence relation">
+    <br>
+</div> -->
````
Same comment

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/lucasf/README.md:
```diff
@@ -0,0 +1,252 @@
+
+#### lucasf( n )
+
+Computes the nth [Lucas number][lucas-number] in single-precision floating-point format.
```
Same comment

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/lucasf/README.md:
````diff
@@ -0,0 +1,252 @@
+<!-- eslint no-undef: "error" -->
+
+```javascript
+var lucasf = require( '@stdlib/math/base/special/lucasf' );
````
This should also be refactored to use `logEachMap`

hrshya replied:
Shouldn't we use `array/base/linspace` here to get the lucasf values in an increasing order like `lucasf( 0 )` till n as compared to using a `discreteUniform` here which would give us random lucasf values.

cc: @anandkaranubc

anandkaranubc replied:
@hrshya Either is fine, I believe. The goal is to showcase a working code example. I changed it to use `discreteUniform` to align with the other refactoring being done in `math/base/special`. Another reason is that you were using `linspace` with four arguments, including `opts`, which `linspace` doesn't require. 

https://github.com/stdlib-js/stdlib/blob/18710227bb4adc73c7712fe97930adf933ad4d57/lib/node_modules/%40stdlib/array/base/linspace/lib/main.js#L35

If you think this change was accidentally propagated to other packages, a PR is probably needed for that.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/lucasf/README.md:
```diff
@@ -0,0 +1,252 @@
+
+#### stdlib_base_lucasf( n )
+
+Computes the nth [Lucas number][lucas-number] in single-precision floating-point format.
```
Same comment

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/lucasf/README.md:
````diff
@@ -0,0 +1,252 @@
+Computes the nth [Lucas number][lucas-number] in single-precision floating-point format.
+
+```c
+double out = stdlib_base_lucasf( 0 );
````
```suggestion
float out = stdlib_base_lucasf( 0 );
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/lucasf/README.md:
````diff
@@ -0,0 +1,252 @@
+-   **n**: `[in] int32_t` input value.
+
+```c
+float stdlib_base_lucas( const int32_t n );
````
```suggestion
float stdlib_base_lucasf( const int32_t n );
```

anandkaranubc reviewed (changes requested):
Hi @hrshya, thanks for your work on this! I’ve added some comments on the PR. A few areas need improvement.

hrshya committed:
Update lib/node_modules/@stdlib/math/base/special/lucasf/src/addon.c

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

hrshya committed:
Update lib/node_modules/@stdlib/math/base/special/lucasf/src/main.c

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

hrshya committed:
Update lib/node_modules/@stdlib/math/base/special/lucasf/examples/c/example.c

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

hrshya committed:
Update lib/node_modules/@stdlib/math/base/special/lucasf/lib/main.js

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

hrshya committed:
Update lib/node_modules/@stdlib/math/base/special/lucasf/lib/index.js

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

hrshya committed:
Update lib/node_modules/@stdlib/math/base/special/lucasf/lib/native.js

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

hrshya committed:
Update lib/node_modules/@stdlib/math/base/special/lucasf/README.md

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

hrshya committed:
Update lib/node_modules/@stdlib/math/base/special/lucasf/README.md

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

hrshya committed:
Update lib/node_modules/@stdlib/math/base/special/lucasf/README.md

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

hrshya committed:
Update lib/node_modules/@stdlib/math/base/special/lucasf/src/main.c

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

hrshya committed:
Update lib/node_modules/@stdlib/math/base/special/lucasf/src/main.c

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

hrshya committed:
Update lib/node_modules/@stdlib/math/base/special/lucasf/src/main.c

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

hrshya committed:
Update lib/node_modules/@stdlib/math/base/special/lucasf/package.json

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

hrshya committed:
Update lib/node_modules/@stdlib/math/base/special/lucasf/docs/repl.txt

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

hrshya committed:
Update lib/node_modules/@stdlib/math/base/special/lucasf/benchmark/c/native/benchmark.c

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

hrshya committed:
Update lib/node_modules/@stdlib/math/base/special/lucasf/benchmark/c/native/benchmark.c

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

hrshya committed:
Update lib/node_modules/@stdlib/math/base/special/lucasf/benchmark/benchmark.js

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

hrshya committed:
Update lib/node_modules/@stdlib/math/base/special/lucasf/benchmark/c/native/benchmark.c

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

hrshya committed:
Update lib/node_modules/@stdlib/math/base/special/lucasf/benchmark/c/benchmark.c

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

hrshya committed:
Merge branch 'feat/lucasf' of https://github.com/hrshya/stdlib into feat/lucasf

hrshya committed:
fix: fixes suggested changes

hrshya commented:
@anandkaranubc can you help me out here.

anandkaranubc commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into feat/lucasf

anandkaranubc commented:
/stdlib lint-autofix

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/lucasf/benchmark/benchmark.js:
```diff
@@ -0,0 +1,306 @@
+	x = discreteUniform( 100, 0, 34, {
+		'dtype': 'float32'
+	});
+
+		// TODO: replace with `powf` when available
+		return roundf( pow( PHI, n ) );
+	}
```
@hrshya, this needs to be fixed. There is a missing function implementation. This is the primary reason why the CI/CD workflows are failing. It's a good practice to run the benchmarks locally first to confirm if nothing is broken.

hrshya replied:
I did ran the benchmarks when I first created the PR but not this time because there were any major changes in those files.

anandkaranubc commented:
As a side note, always make sure your feature branch is in sync with the `develop` branch. This was the reason why there were many unrelated file changes in your previous commits.

[Discussion](https://github.com/stdlib-js/stdlib/blob/develop/docs/contributing/FAQ.md#auto-push)

hrshya commented:
> As a side note, always make sure your feature branch is in sync with the `develop` branch. This was the reason why there were many unrelated file changes in your previous commits.
> 
> [Discussion](https://github.com/stdlib-js/stdlib/blob/develop/docs/contributing/FAQ.md#auto-push)

Actually it was in sync but the thing was I didn't have this branch locally so I pulled it from origin and i thought as my develop branch is in sync so this branch would also be but i was wrong and i should have checked it before commiting

hrshya committed:
adds function implementation

Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

hrshya committed:
update benchmark.js

Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

hrshya committed:
resolve lint issue

Signed-off-by: Harsh <149176984+hrshya@users.noreply.github.com>

hrshya commented:
@anandkaranubc All the issues have been resolved. Can you give it a quick review.

anandkaranubc commented:
/stdlib merge

anandkaranubc committed:
docs: fix function descriptions and example scripts

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
chore: clean-up

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
bench: follow code conventions

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
chore: re-enable lint rule

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
chore: update function description

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
test: update test messages

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
docs: add placeholder comments for related links

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc reviewed (approved):
Good to go! Thanks @hrshya.
